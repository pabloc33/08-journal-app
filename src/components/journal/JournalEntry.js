import React from "react";

export const JournalEntry = () => {
  return (
    <div className="journal__entry pointer">
      <div
        className="journal__entry-picture"
        style={{
          backgroundSize: "cover",
          backgroundImage:
            "url(https://i.pinimg.com/originals/67/54/78/675478c7dcc17f90ffa729387685615a.jpg)",
        }}
      ></div>
      <div className="journal__entry-body">
        <p className="journal__entry-title">Un nuevo mundo</p>
        <p className="journal__entry-content">
          Lorem ipsum dolor, sit amet consectetur adipisicing elit.
        </p>
      </div>
      <div className="journal__entry-date-box">
        <span>Monday</span>
        <h4>28</h4>
      </div>
    </div>
  );
};
