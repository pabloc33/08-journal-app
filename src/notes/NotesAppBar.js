import React from "react";

export const NotesAppBar = () => {
  return (
    <div className="notes__appbar">
      <samp>24 de septiembre de 2020</samp>

      <div>
        <button className="btn">Picture</button>
      </div>

      <div>
        <button className="btn">Save</button>
      </div>
    </div>
  );
};
