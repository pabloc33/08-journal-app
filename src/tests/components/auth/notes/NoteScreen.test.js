import React from "react";
import { Provider } from "react-redux";
import configureStore from "redux-mock-store"; //ES6 modules
import thunk from "redux-thunk";
import "@testing-library/jest-dom";
import { Sidebar } from "../../../../components/journal/Sidebar";
import { activeNotes } from "../../../../actions/notes";
import { NoteScreen } from "../../../../notes/NoteScreen";
const { mount } = require("enzyme");

jest.mock("../../../../actions/notes", () => ({
  activeNotes: jest.fn(),
}));

const middlewares = [thunk];
const mockStore = configureStore(middlewares);

const initState = {
  auth: {
    uid: "1",
    name: "Pablo",
  },
  ui: {
    loading: false,
    msgError: null,
  },
  notes: {
    active: {
      id: 1234,
      title: "Hola",
      body: "mundo",
      date: 0,
    },
    notes: [],
  },
};

let store = mockStore(initState);
store.dispatch = jest.fn();

const wrapper = mount(
  <Provider store={store}>
    <NoteScreen />
  </Provider>
);

describe("Pruebas en el <NoteScreen />", () => {
  test("debe de mostrarse correctamente ", () => {
    expect(wrapper).toMatchSnapshot();
  });

  test("debe de disparar el active note ", () => {
    wrapper.find('input[name="title"]').simulate("change", {
      target: {
        name: "title",
        value: "Hola de nuevo",
      },
    });

    expect(activeNotes).toHaveBeenLastCalledWith(1234, {
      body: "mundo",
      title: "Hola de nuevo",
      id: 1234,
      date: 0,
    });
  });
});
