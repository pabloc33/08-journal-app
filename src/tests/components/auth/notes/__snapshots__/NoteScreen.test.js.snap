// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en el <NoteScreen /> debe de mostrarse correctamente  1`] = `
<div
  className="notes__main-content"
>
  <div
    className="notes__appbar"
  >
    <samp>
      24 de septiembre de 2020
    </samp>
    <input
      id="fileSelector"
      name="file"
      onChange={[Function]}
      style={
        Object {
          "display": "none",
        }
      }
      type="file"
    />
    <div>
      <button
        className="btn"
        onClick={[Function]}
      >
        Picture
      </button>
    </div>
    <div>
      <button
        className="btn"
        onClick={[Function]}
      >
        Save
      </button>
    </div>
  </div>
  <div
    className="notes__content"
  >
    <input
      autoComplete="off"
      className="notes__title-input"
      name="title"
      onChange={[Function]}
      placeholder="Some awesome title"
      type="text"
      value="Hola"
    />
    <textarea
      className="notes__textarea"
      name="body"
      onChange={[Function]}
      placeholder="What happened today"
      value="mundo"
    />
  </div>
  <button
    className="btn btn-danger"
    onClick={[Function]}
  >
    Delete
  </button>
</div>
`;
